Tests/ApiEndpoints/test_allhousingapp_pytest.py

```diff
@@ -82,6 +81,68 @@ def test_not_on_whitelist(self):
             pytest.fail('Expected 404 Not Found, got {0}.'\
                 .format(response.status_code))
 
+#    Verify that an application and all rows that reference are deleted successfully
+#    Endpoint -- 'api/housing/apartment/applications/{applicationID}'
+#    Expected Status Code -- 200 OK and 404 Not Found
+#    Expected Content -- Empty Response content
+    def test_application_deleted(self):
+        self.session = self.createAuthorizedSession(control.username, control.password)
+        self.url = control.hostURL + 'api/housing/apartment/applications'
+        self.data = {
+            'ApplicationID' : -1,
+            'EditorUsername' : control.leader_username,
+            'Applicants' : [
+                {
+                    'Username' : control.leader_username
+                },
+                {
+                    'Username' : control.username
+                }
+            ],
+            'ApartmentChoices' : [
+                {
+                    'HallRank' : 1,
+                    'HallName' : "Taj MaHall"
+                },
+                {
+                    'HallRank' : 2,
+                    'HallName' : "Carnegie Hall"
+                },
+                {
+                    'HallRank' : 3,
+                    'HallName' : "Hall of Mirrors"
+                }
+            ],
```

Comment:
Will need to update the stored procedure to check that the halls are actually things in the AA_ApartmentHalls table, so unfortunately we will eventually have to change these hall names